pkg/kubelet/stats/cri_stats_provider.go

```diff
@@ -826,7 +826,29 @@ func getCRICadvisorStats(infos map[string]cadvisorapiv2.ContainerInfo) map[strin
 		if !isPodManagedContainer(&info) {
 			continue
 		}
-		stats[path.Base(key)] = info
+		stats[extractIDFromCgroupPath(key)] = info
 	}
 	return stats
 }
+
+func extractIDFromCgroupPath(cp string) string {
```

Comment:
Any reason not to use the full name here?

```suggestion
func extractIDFromCgroupPath(cgroupPath string) string {
```

Reply:
@ehashman  Changed as suggested.  And also fix test to generate containerdID by UUID without dash